src/Engine/ProtoCore/CompilerUtils.cs

```diff
@@ -252,9 +254,10 @@ public static bool TryLoadAssemblyIntoCore(Core core, string assemblyPath)
         /// stores list of AST nodes, errors and warnings
         /// Evaluates and stores list of unbound identifiers
         /// </summary>
-        /// <param name="parseParams"></param>
-        /// <returns></returns>
-        public static bool PreCompileCodeBlock(Core core, ref ParseParam parseParams)
+        /// <param name="parseParams"> container for compilation related parameters </param>
+        /// <param name="elementResolver"> classname resolver </param>
+        /// <returns> true if code compilation succeeds, false otherwise </returns>
+        public static bool PreCompileCodeBlock(Core core, ref ParseParam parseParams, ElementResolver elementResolver)
```

Comment:
Please do not change the signature of `PreCompileCodeBlock` if possible. The decision to use CBN-specific resolver or workspace resolver should be decided in the callers (the `parseParams` should carry the right `ElementResolver` the moment it gets here (it's not the job of `PreCompileCodeBlock` method to decide). Please let me know if that's not possible.